src/app/features/users-page/users-page.component.ts

```diff
@@ -1,25 +1,74 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
+import { FilterModel } from '@features/filter-bar/interfaces/filter-model';
 import { IUsers } from '@features/users-page/interfaces/users';
 import { getUsersPending } from '@features/users-page/store/users.actions';
 import { getUsersFromState } from '@features/users-page/store/users.selectors';
 import { Store } from '@ngrx/store';
 import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Component({
+    changeDetection: ChangeDetectionStrategy.OnPush,
     selector: 'crx-users-page',
     styleUrls: ['./users-page.component.scss'],
     templateUrl: './users-page.component.html'
 })
 
-export class UsersPageComponent {
+export class UsersPageComponent implements OnInit {
 
+    @Input() filters: any;
+
+    public filterIsOpen = false;
     public users$: Observable<IUsers[]>;
 
     constructor (private store: Store) {
 
         this.store.dispatch(getUsersPending());
+
+    }
+
+    ngOnInit () {
+
         this.users$ = this.store.select(getUsersFromState);
 
     }
 
+    openFilter () {
+
+        return this.filterIsOpen = !this.filterIsOpen;
+
+    }
+
+    applyFilters (event: FilterModel) {
+
+        if (event.byGender) {
+
+            this.users$ = this.users$.pipe(map((users) =>
+                users?.filter((user) => user.gender === event.byGender)));
+
+        } else if (event.byState) {
+
+            this.users$ = this.users$.pipe(map((users) =>
+                users?.filter((user) => user.state === event.byState)));
+
+        } else if (event.byDate.value.start && event.byDate.touched) {
```

Comment:
parent element shouldn't know about form controls structure. 
your `FilterModel` should provide 
```
birthDateFrom: Date;
birthDateTo: Date;
```

just values, all the logic about form handling should be encapsulated inside a component with a form